core/domain/prod_validation_jobs_one_off.py

```diff
@@ -40,13 +42,53 @@ def _external_id_relationships(cls):
             model to validate, and whose values are the corresponding model
             classes.
         """
-        raise NotImplementedError
+        return {}
+
+    @classmethod
+    def _custom_external_id_relationship(cls, item):  # pylint: disable=unused-argument
+        """Define custom external relationships.
+
+        Sometimes we need more complicated logic to determine which external
```

Comment:
On first reading I didn't understand the connection of this paragraph to this method. Maybe be more explicit about that? 

(I tried to write an example but then I realized that I didn't understand the difference between this and _external_id_relationships.)

Reply:
Let me know if you think generalizing makes sense.

In _external_id_relationships we look at some field of the item the value of the field is the key we use to check external models.
But this doesn't address all use cases. For example, the value to check for the state id mapping model is {exp_id}.{exp_version} so we need to have a way to specify these kind of external model relationships.

Reply:
Oh, I see.

I could go either way. I would say it depends on how much _custom_external_id_relationships() will end up being used, given the model checks we have. If we would end up using it a lot, then I think that having one method might be better. If we don't, then having two methods sounds reasonable.

Reply:
Oh ok, in that case I'll keep them different methods for now.

Reply:
Actually ... looking at the current file, I think I am leaning much more strongly now towards just making them one method. I don't see a strong need to introduce a "standard" + "extension" layer here, especially since the "standard" part is not defined declaratively. Can we combine them? That would also simplify the naming and documentation in e.g. _fetch_external_models (which actually checks both methods, not just _get_external_id_relationships()).

Reply:
Done!